SparseGrids/gridtestExternalTests.cpp

```diff
@@ -335,108 +406,117 @@ bool ExternalTester::performGlobalTest(TasGrid::TypeOneDRule rule) const{
     int wfirst = 10, wsecond = 35, wthird = 15;
     if (rule == TasGrid::rule_clenshawcurtis){
         { TasGrid::TypeOneDRule oned = TasGrid::rule_clenshawcurtis;
-        const int depths1[3] = { 25, 25, 25 };
-        const double tols1[3] = { 1.E-12, 1.E-12, 1.E-11 };
-        const int depths2[3] = { 25, 27, 27 };
-        const double tols2[3] = { 1.E-12, 1.E-10, 1.E-11 };
-        if (testGlobalRule(&f21nx2, oned, 0, alpha, beta, true, depths1, tols1) && testGlobalRule(&f21cos, oned, 0, alpha, beta, true, depths2, tols2)){
+        const int depths1[4] = { 25, 25, 25, 25 };
+        const double tols1[4] = { 1.E-12, 1.E-12, 1.E-11, 1.E-10 };
+        const int depths2[4] = { 25, 27, 27, 27 };
+        const double tols2[4] = { 1.E-12, 1.E-10, 1.E-11, 1.E-10 };
+        if (testGlobalRule(&f21nx2, oned, 0, alpha, beta, all_test_types, depths1, tols1) &&
+            testGlobalRule(&f21cos, oned, 0, alpha, beta, all_test_types, depths2, tols2)){
             if (verbose) cout << setw(wfirst) << "Rule" << setw(wsecond) << IO::getRuleString(oned) << setw(wthird) << "Pass" << endl;
         }else{
             cout << setw(wfirst) << "Rule" << setw(wsecond) << IO::getRuleString(oned) << setw(wthird) << "FAIL" << endl;  pass = false;
         }}
     }else if (rule == TasGrid::rule_clenshawcurtis0){
         { TasGrid::TypeOneDRule oned = TasGrid::rule_clenshawcurtis0;
-        const int depths1[3] = { 25, 25, 25 };
-        const double tols1[3] = { 1.E-12, 1.E-12, 1.E-11 };
-        if (testGlobalRule(&f21sinsin, oned, 0, alpha, beta, true, depths1, tols1)){
+        const int depths1[4] = { 25, 25, 25, 25 };
+        const double tols1[4] = { 1.E-12, 1.E-12, 1.E-11, 1.E-10 };
+        if (testGlobalRule(&f21sinsin, oned, 0, alpha, beta, all_test_types, depths1, tols1)){
             if (verbose) cout << setw(wfirst) << "Rule" << setw(wsecond) << IO::getRuleString(oned) << setw(wthird) << "Pass" << endl;
         }else{
             cout << setw(wfirst) << "Rule" << setw(wsecond) << IO::getRuleString(oned) << setw(wthird) << "FAIL" << endl;  pass = false;
         }}
     }else if ((rule == TasGrid::rule_chebyshev) || (rule == TasGrid::rule_chebyshevodd)){
         { TasGrid::TypeOneDRule oned = rule;
-        const int depths1[3] = { 22, 22, 22 };
-        const double tols1[3] = { 1.E-12, 1.E-10, 1.E-10 };
-        const int depths2[3] = { 22, 22, 22 };
-        const double tols2[3] = { 1.E-12, 1.E-09, 1.E-09 };
-        if (testGlobalRule(&f21nx2, oned, 0, alpha, beta, true, depths1, tols1) && testGlobalRule(&f21cos, oned, 0, alpha, beta, true, depths2, tols2)){
+        const int depths1[4] = { 22, 22, 22, 22 };
+        const double tols1[4] = { 1.E-12, 1.E-10, 1.E-10, 1.E-09 };
+        const int depths2[4] = { 22, 22, 22, 22 };
+        const double tols2[4] = { 1.E-12, 1.E-09, 1.E-09, 1.E-08 };
+        if (testGlobalRule(&f21nx2, oned, 0, alpha, beta, all_test_types, depths1, tols1) &&
```

Comment:
It's Ok to break the row, but change `&&` to `and`.

Reply:
Changed.